syntax = "proto3";

package protocols.behavior.uninfication;

import "protocols/common/peripheral_actuation.proto";

import "protocols/navigation/motion.proto";
import "protocols/navigation/navigation_service.proto";
import "protocols/navigation/planning.proto";

/* 
    Remove this implementation after CBR-2024.
    This is a replica of current implementation of ssl-unification behavior output.
*/

message Behavior {
    oneof output {
        Motion motion = 1;
        Planning planning = 2;
        Navigation navigation = 3;
    }
}

message Motion {
    oneof type {
        GoToPoint go_to_point = 1;
        GoToPointWithTrajectory go_to_point_with_trajectory = 2;
        RotateInPoint rotate_in_point = 3;
        RotateOnSelf rotate_on_self = 4;
    }
    PeripheralActuation peripheral_actuation = 2;
}

